sklearn/ensemble/ensemble_selection.py

```diff
@@ -0,0 +1,261 @@
+        If None, the random number generator is the RandomState instance used
+        by `np.random`.
+
+    Attributes
+    ----------
+    ensemble_ : counter of classifiers
+        The collection of fitted estimators and its weight.
+
+    classes_ : array-like, shape = [n_classes_]
+        The classes labels.
+
+    n_classes_ : int
+        The number of classes.
+
+    References
+    ----------
+    .. [1] R. Caruana, A. Niculescu-Mizil, G. Crew, A. Ksikes, "Ensemble
+           Selection from Libraries of Models", 2004.
+
+    .. [2] R. Caruana, A. Munson, A. Niculescu-Mizil, "Getting the Most Out
+           of Ensemble Selection", 2006.
+
+    """
+
+    _metrics = {
+        'f1': _f1,
+        'accuracy': _accuracy,
+    }
+
+    def __init__(self, estimators=None, max_bag_estimators=50, bag_fraction=0.5,
+                 n_bags=20, n_best=1, score_metric='accuracy',
+                 prune_fraction=0, random_state=None, verbose=False):
+        self.estimators = estimators
+        self.named_estimators = dict(estimators)
+        self.max_bag_estimators = max_bag_estimators
+        self.bag_fraction = bag_fraction
+        self.n_bags = n_bags
+        self.n_best = n_best
+        self.score_metric = score_metric
+        self.prune_fraction = prune_fraction
+        self.random_state = random_state
+        self.verbose = verbose
+
+    def fit(self, X, y):
+        """ Conduct ensemble selection on the validation set (X, y).
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Validation vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        Returns
+        -------
+        self : object
+        """
+        if isinstance(y, np.ndarray) and len(y.shape) > 1 and y.shape[1] > 1:
+            raise NotImplementedError('Multilabel and multi-output'
+                                      ' classification is not supported.')
+
+        metric_names = self._metrics.keys()
+        if self.score_metric not in metric_names:
+            raise ValueError('score_metric %s not in %s' %
+                             (self.score_metric, metric_names))
+
+        if self.estimators is None or len(self.estimators) == 0:
+            raise AttributeError('Invalid `estimators` attribute, `estimators`'
+                                 ' should be a list of'
+                                 ' (string, fitted_estimator) tuples')
+
+        self.score_metric_ = self._metrics[self.score_metric]
+
+        # store each clf's score and predict_proba result on validation set
+        self.clfs_score_ = {}
+        self.clfs_proba_ = {}
+        for (name, clf) in self.estimators:
+            score, proba = self._get_score_of_model(X, y, clf)
+            self.clfs_score_[name] = score
+            self.clfs_proba_[name] = proba
```

Comment:
I don't think this data should be stored on the model if it is not used during prediction.